cirq/sim/density_matrix_simulator_test.py

```diff
@@ -936,6 +936,19 @@ def test_random_seed_does_not_modify_global_state_non_terminal_measurements():
     assert result1 == result2
 
 
+def test_random_seed_non_terminal_measurements_deterministic():
```

Comment:
Increase the number of repetitions to 30. We don't want false negatives.

Reply:
Done.